src/helpers/generate-matrix.ts

```diff
@@ -11,17 +11,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-let */
+
 import { HelperInputs } from '../types/generated';
 import { chunk } from 'lodash';
 
 export class GenerateMatrix extends HelperInputs {
   paths = '';
   batches?: string;
+  load_balancing_sizes?: string;
 }
 
-export const generateMatrix = ({ paths, batches = '1' }: GenerateMatrix) => {
+export const generateMatrix = ({ paths, batches: _batches = '1', load_balancing_sizes }: GenerateMatrix) => {
   const matrixValues = paths.split(/[\n,]/);
+  const batches = Number(_batches);
+  if (!load_balancing_sizes) {
+    return {
+      include: chunk(matrixValues, Math.ceil(matrixValues.length / batches)).map(chunk => ({ path: chunk.join(',') }))
+    };
+  }
+  const loadBalancingSizes = load_balancing_sizes.split(/[\n,]/).map(size => Number(size));
+  if (loadBalancingSizes.length !== matrixValues.length)
+    throw new Error('load_balancing_sizes input must have the same length as paths input');
+  const targetLoadSize = loadBalancingSizes.reduce((acc, size) => acc + size, 0) / batches;
```

Comment:
Nit pick - we could probably use the `sum` function from lodash to make this slightly more readable

Reply:
sure that makes sense